### ISSUE TITLE
Slow training, does not utilize GPUs

### ISSUE THREAD

**Issue by plbecker** [matterport/Mask_RCNN#1834]
Hi!
Awesome library first of all!

I am having an issue running the training code / a rewritten version of the balloon example for custom data on my server. I am transfer learning around 1000 images of Full HD resolution on a DGX-1 (8x V100 GPUs with 32GB VRAM each). 
All drivers are installed correctly, Tensorflow-GPU has version 1.14.0 (non-GPU TF is not installed in the virtualenv), Keras has 2.2.5 due to the issue #1754.
The training works fine, albeit very slow (it takes around 50min per epoch) and the GPUs are not running much at all (all is done by the two 20-core Intel Xeon E5-2698) which is not very efficient. When running `nvidia-smi`, it does report the 8 jobs it is supposed to have, but every GPU has 0% utilization and only uses 418MB out of 32GB VRAM.
During training, it uses 400GB - 500GB of RAM

Is there any way to enable GPU training / any possibilty it is disabled by default? I am running the parameters: `IMAGES_PER_GPU=4` with 100 steps per epoch and GPU_COUNT=8

**Comment by Dayan-Zhanchi** [comment]
+1 I have the same problem with the GPU not running (I only have a nvidia rtx2060 though). In my case it seems to not run at all for some reason (GPU usage stays around pretty much 0%). Tensorflow (tensorflow-gpu 1.14) can detect my GPU so I don't know what the problem is.

**Comment by plbecker** [comment]
So I did some other tests, apparently it was an issue of some CUDA-related libraries, not anything related to Mask R-CNN. I didn't get any errors and Tensorflow-GPU seemed to be running, but internally it had some issues. I fixed it by using a slightly modified Nvidia-Docker image specifically for the DGX-1 which worked directly without any issues (per epoch, the system now needed ~5min instead of 50+). I suggest using a docker image with pre-installed CUDA, cudNN and TF tailored for the specific system you are using, especially when it comes to supercomputers

**Comment by saskra** [comment]
I have the same problem but could not solve it as you described. It only happens with Mask_RCNN, not with other packages. And even with Mask_RCNN sometimes the GPUs work, mostly they do not - I could not find any pattern.

**Comment by christianburbon** [comment]
I'm having the same problem as this. However, it only happens when I do training on a dataset that has a lot of segmentation points.

My baseline comparison was training https://www.kaggle.com/alexo98/leaf-detection. Which is able to utilize my GPU, and training is very fast. (however segmentation points here were only bbox)

I've already tried resizing my image prior to training which reduced image file size from 20MB to 400KB. segmentation was resized as well.

I'm using only layers = 3+

Hardware: NVIDIA GeForce GTX 1050 Ti

Can anyone suggest solutions?
